### ISSUE TITLE
Add support for hints in notification portal

### ISSUE THREAD

**Issue by grulja** [flatpak/xdg-desktop-portal#77]
Add support for: https://developer.gnome.org/notification-spec/#hints

That way we should be able to pass custom properties through the notification portal. E.g. in KDE we use many custom properties that are just passed in "hints" as a simple variant map. It shouldn't be hard to support that, this might just be one more specified option in "notification" parameter in AddNotification method.  

E.g. hints a{sv} - array of extra properties, the backend implementation is not required to support any of them

**Comment by matthiasclasen** [comment]
the fundamental problem with such optional hints is that apps can't rely on them, and don't know what the user experience will be.

**Comment by TingPing** [comment]
I don't think that is a real problem, the fdo interface allows the client to query supported features.

**Comment by Mikenux** [comment]
There is a new proposal to discuss for notifications: https://github.com/flatpak/xdg-desktop-portal/issues/983

**Comment by RokeJulianLockhart** [comment]
> https://github.com/flatpak/xdg-desktop-portal/issues/77#issue-194942025

@grulja, https://developer.gnome.org/notification-spec/#hints has been replaced by https://developer-old.gnome.org/notification-spec/#hints.
